spec/app/domain/authentication/authn_k8s/execute_command_in_container_spec.rb

```diff
@@ -169,6 +164,65 @@ def subject_in_thread(ws_client:, timeout:, body:, stdin:)
   end
 
   context "Calling ExecuteCommandInContainer" do
+    context "converts endpoint for websocket client" do
+      let(:kube_client_api_endpoint) do
+        raise "@kube_client_api_endpoint not defined" unless @kube_client_api_endpoint
+
+        URI.parse(@kube_client_api_endpoint)
+      end
+
+      let(:ws_client) do 
+        ws_client = WsClientMock.new(handshake_error: nil)
+        
+        thread = subject_in_thread(
+          ws_client: ws_client,
+          timeout: 1,
+          body: body,
+          stdin: true
+        )
+
+        ws_client.trigger_open
+        ws_client.trigger_message(message)
+        ws_client.trigger_close
+        thread.join
+        thread[:output]
+
+        ws_client
+      end
+
+      it "retains subpath" do
+        @kube_client_api_endpoint = "https://path/to"
```

Comment:
Could the endpoint ever be defined with IPv4/IPv6 addresses here?

Reply:
It definitely could, and it should just work since we're representing the URI using this module https://docs.ruby-lang.org/en/2.1.0/URI/Generic.html. The host can be IPv4/IPv6 addresses, hostnames etc.

Reply:
Some table tests would might be good to cover this sort of thing